dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h

```diff
@@ -250,8 +251,46 @@ class DMFilePackFilter
                     read_limiter);
                 return MinMaxIndex::read(*type, index_buf, index_file_size);
             }
-            else
+            else if (dmfile->useMetaV2() and !DB::S3::ClientFactory::instance().isEnabled()) // v3
```

Comment:
I think we can't use the similar logic like s3 in newRandomAccessFile to decide which branch we use. It will make the newRandomAccessFile quite strange.

Reply:
done